<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Guidelines for bioconda recipes &#8212; Bioconda  documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css?v=39de963d" />
    <link rel="stylesheet" type="text/css" href="../_static/graphviz.css?v=fd3f3429" />
    <link rel="stylesheet" type="text/css" href="../_static/style.css?v=48720263" />
    <link rel="stylesheet" type="text/css" href="../_static/font-awesome-4.7.0/css/font-awesome.min.css?v=02f538dc" />
    <script src="../_static/documentation_options.js?v=5929fcd5"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/vega.min.js?v=d54ef115"></script>
    <script src="../_static/vega-lite.min.js?v=4fab2bf4"></script>
    <script src="../_static/vega-embed.min.js?v=2f6bc6e4"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Linting" href="linting.html" />
    <link rel="prev" title="Build system" href="build-system.html" />
    <link href="https://fonts.googleapis.com/css?family=Lato|Raleway" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Raleway:wght@700&display=swap" rel="stylesheet">


    <link rel="apple-touch-icon" sizes="57x57" href="../_static/apple-icon-57x57.png">
    <link rel="apple-touch-icon" sizes="60x60" href="../_static/apple-icon-60x60.png">
    <link rel="apple-touch-icon" sizes="72x72" href="../_static/apple-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="76x76" href="../_static/apple-icon-76x76.png">
    <link rel="apple-touch-icon" sizes="114x114" href="../_static/apple-icon-114x114.png">
    <link rel="apple-touch-icon" sizes="120x120" href="../_static/apple-icon-120x120.png">
    <link rel="apple-touch-icon" sizes="144x144" href="../_static/apple-icon-144x144.png">
    <link rel="apple-touch-icon" sizes="152x152" href="../_static/apple-icon-152x152.png">
    <link rel="apple-touch-icon" sizes="180x180" href="../_static/apple-icon-180x180.png">
    <link rel="icon" type="image/png" sizes="192x192"  href="../_static/android-icon-192x192.png">
    <link rel="icon" type="image/png" sizes="32x32" href="../_static/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="96x96" href="../_static/favicon-96x96.png">
    <link rel="icon" type="image/png" sizes="16x16" href="../_static/favicon-16x16.png">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="msapplication-TileImage" content="../_static/ms-icon-144x144.png">

    <link rel="manifest" href="../_static/manifest.json">
    <meta name="theme-color" content="#ffffff">

   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  

  
  


  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="guidelines-for-bioconda-recipes">
<span id="guidelines"></span><h1>Guidelines for <code class="docutils literal notranslate"><span class="pre">bioconda</span></code> recipes<a class="headerlink" href="#guidelines-for-bioconda-recipes" title="Link to this heading">¶</a></h1>
<section id="bioconda-recipe-checklist">
<h2>bioconda recipe checklist<a class="headerlink" href="#bioconda-recipe-checklist" title="Link to this heading">¶</a></h2>
<ul class="simple">
<li><p>Source URL is stable (<a class="reference internal" href="#stable-url"><span class="std std-ref">details</span></a>)</p></li>
<li><p>sha256 or md5 hash included for source download (<a class="reference internal" href="#hashes"><span class="std std-ref">details</span></a>)</p></li>
<li><p>Appropriate build number (<a class="reference internal" href="#buildnum"><span class="std std-ref">details</span></a>)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">.bat</span></code> file for Windows removed (<a class="reference internal" href="#bat-files"><span class="std std-ref">details</span></a>)</p></li>
<li><p>Remove unnecessary comments (<a class="reference internal" href="#comments-in-meta"><span class="std std-ref">details</span></a>)</p></li>
<li><p>Adequate tests included (<a class="reference internal" href="#tests"><span class="std std-ref">details</span></a>)</p></li>
<li><p>Files created by the recipe follow the FSH (<a class="reference internal" href="#fsh-section"><span class="std std-ref">details</span></a>)</p></li>
<li><p>License allows redistribution and license is indicated in <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code></p></li>
<li><p>Package does not already exist in the <code class="docutils literal notranslate"><span class="pre">defaults</span></code> or <code class="docutils literal notranslate"><span class="pre">conda-forge</span></code>
channels with some exceptions (<a class="reference internal" href="#channel-exceptions"><span class="std std-ref">details</span></a>)</p></li>
<li><p>Package is appropriate for bioconda (<a class="reference internal" href="#appropriate-for-bioconda"><span class="std std-ref">details</span></a>)</p></li>
<li><p>If the recipe installs custom wrapper scripts, usage notes should be added to
<code class="docutils literal notranslate"><span class="pre">extra</span> <span class="pre">-&gt;</span> <span class="pre">notes</span></code> in the <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code>.</p></li>
<li><p><strong>Update 21 Jan 2019</strong>:  Recipes that contain pure Python packages should be marked as a <a class="reference external" href="https://www.continuum.io/blog/developer-blog/condas-new-noarch-packages">“noarch”</a>
(<a class="reference internal" href="#id3"><span class="std std-ref">details</span></a>).</p></li>
<li><p><strong>Update 7 Mar 2018</strong>: When patching a recipe, please provide details on how
you tried to address the problem upstream (<a class="reference internal" href="#patching"><span class="std std-ref">details</span></a>)</p></li>
</ul>
<section id="stable-urls">
<span id="stable-url"></span><h3>Stable urls<a class="headerlink" href="#stable-urls" title="Link to this heading">¶</a></h3>
<p>While supported by conda, <code class="docutils literal notranslate"><span class="pre">git_url</span></code> and <code class="docutils literal notranslate"><span class="pre">git_rev</span></code> are not as stable as a git
tarball. Ideally, a github repo should have tagged releases that are accessible
as tarballs from the “releases” section of the github repo. Correspondingly, a
bitbucket repo should have have tagged versions that are accessible as tarballs
from the “Downloads” -&gt; “tags” section of the bitbucket repo.</p>
<p>TODO: additional info on the various R and bioconductor URLs</p>
</section>
<section id="hashes">
<span id="id1"></span><h3>Hashes<a class="headerlink" href="#hashes" title="Link to this heading">¶</a></h3>
<p>We support either sha256 or md5 checksums to verify the integrity of the source
package. If a checksum is provided alongside the source package, then use that.
Otherwise we prefer sha256 over md5.</p>
<p>Use <code class="docutils literal notranslate"><span class="pre">shasum</span> <span class="pre">-a</span> <span class="pre">256</span> <span class="pre">...</span></code> (or <code class="docutils literal notranslate"><span class="pre">sha256sum</span></code>  or <code class="docutils literal notranslate"><span class="pre">openssl</span> <span class="pre">sha256</span></code> etc) on a
file to compute its sha256 hash, and copy this into the recipe, for example:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-O-<span class="w"> </span><span class="nv">$URL</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>shasum<span class="w"> </span>-a<span class="w"> </span><span class="m">256</span>
</pre></div>
</div>
<p>Likewise use <code class="docutils literal notranslate"><span class="pre">md5sum</span></code> (Linux) or <code class="docutils literal notranslate"><span class="pre">md5</span></code> (OSX) on a file to compute its md5 hash,
and copy this into the recipe.</p>
</section>
<section id="build-numbers">
<span id="buildnum"></span><h3>Build numbers<a class="headerlink" href="#build-numbers" title="Link to this heading">¶</a></h3>
<p>The build number (see <code class="xref any docutils literal notranslate"><span class="pre">conda</span> <span class="pre">docs</span></code>) can be
used to trigger a new build for a package whose version has not
changed.  This is useful for fixing errors in recipes. The first
recipe for a new version should always have a build number of 0.</p>
</section>
<section id="bat-files">
<span id="id2"></span><h3><code class="docutils literal notranslate"><span class="pre">.bat</span></code> files<a class="headerlink" href="#bat-files" title="Link to this heading">¶</a></h3>
<p>When creating a recipe using one of the <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span></code> tools, a <code class="docutils literal notranslate"><span class="pre">.bat</span></code>
file for Windows will be created. Since bioconda does not support Windows and
to reduce clutter, please remove these files</p>
</section>
<section id="comments-in-recipes">
<span id="comments-in-meta"></span><h3>Comments in recipes<a class="headerlink" href="#comments-in-recipes" title="Link to this heading">¶</a></h3>
<p>When creating a recipe using one of the <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span></code> tools, often many
comments are included, for example, to point out sections that can be
uncommented and used. Please delete all auto-generated comments in
<code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code> and <code class="docutils literal notranslate"><span class="pre">build.sh</span></code>. But please add any comments that you feel could
help future maintainers of the recipe, especially if there’s something
non-standard.</p>
</section>
<section id="filesystem-hierarchy-standard">
<span id="fsh-section"></span><h3>Filesystem Hierarchy Standard<a class="headerlink" href="#filesystem-hierarchy-standard" title="Link to this heading">¶</a></h3>
<p>Recipes should conform to the Filesystem Hierarchy Standard (<a class="reference external" href="https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard">FSH</a>). This is most
important for libraries and Java packages; for these cases use one of the
recipes below as a guideline.</p>
</section>
<section id="existing-package-exceptions">
<span id="channel-exceptions"></span><h3>Existing package exceptions<a class="headerlink" href="#existing-package-exceptions" title="Link to this heading">¶</a></h3>
<p>If a package already exists in one of the dependent channels but is broken or
cannot be used as-is, please first consider fixing the package in that channel.
If this is not possible, please indicate this in the PR and notify
<code class="docutils literal notranslate"><span class="pre">&#64;bioconda/core</span></code> in the PR.</p>
</section>
<section id="packages-appropriate-for-bioconda">
<span id="appropriate-for-bioconda"></span><h3>Packages appropriate for bioconda<a class="headerlink" href="#packages-appropriate-for-bioconda" title="Link to this heading">¶</a></h3>
<p>Bioconda is a bioinformatics channel, so we prefer to host packages specific to
this domain. If a bioinformatics recipe has more general dependencies, please
consider opening a pull request with <a class="reference external" href="https://conda-forge.github.io/#add_recipe">conda-forge</a> which hosts general packages.</p>
<p>All CRAN packages that do not depend on a package in bioconda should be added
to conda-forge instead. This is still the case if the CRAN package is directly
related to bioinformatics.</p>
<p>If uploading of an unreleased version is necessary, please follow the
versioning scheme of conda for pre- and post-releases (e.g. using a, b, rc, and
dev suffixes, see <a class="reference external" href="https://github.com/conda/conda/blob/d1348cf3eca0f78093c7c46157989509572e9c25/conda/version.py#L30">here</a>).</p>
</section>
<section id="noarch-packages">
<span id="id3"></span><h3>“Noarch” packages<a class="headerlink" href="#noarch-packages" title="Link to this heading">¶</a></h3>
<p>A <a class="reference internal" href="#id3"><span class="std std-ref">“Noarch” packages</span></a> package must be created for pure python packages. To do so,
add <code class="docutils literal notranslate"><span class="pre">noarch:</span> <span class="pre">python</span></code> to the <code class="docutils literal notranslate"><span class="pre">build</span></code> section of the <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code> file.</p>
<dl class="simple">
<dt>For other generic packages (like a data package), add <code class="docutils literal notranslate"><span class="pre">noarch:</span> <span class="pre">generic</span></code> to</dt><dd><p>the <code class="docutils literal notranslate"><span class="pre">build</span></code> section.</p>
</dd>
</dl>
</section>
<section id="dependencies">
<h3>Dependencies<a class="headerlink" href="#dependencies" title="Link to this heading">¶</a></h3>
<p>There is currently no mechanism to define, in the <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code> file, that
a particular dependency should come from a particular channel. This means that
a recipe must have its dependencies in one of the following:</p>
<ul class="simple">
<li><p>as-yet-unbuilt recipes in the repo but that will be included in the PR</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">conda-forge</span></code> channel</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">bioconda</span></code> channel</p></li>
<li><p>default Anaconda channel</p></li>
</ul>
<p>Otherwise, you will have to write the recipes for those dependencies and
include them in the PR. One shortcut is to use <code class="docutils literal notranslate"><span class="pre">anaconda</span> <span class="pre">search</span> <span class="pre">-t</span> <span class="pre">conda</span>
<span class="pre">&lt;dependency</span> <span class="pre">name&gt;</span></code> to look for other packages built by others. Inspecting those
recipes can give some clues into building a version of the dependency for
bioconda.</p>
</section>
<section id="test-dependencies">
<span id="id4"></span><h3>Test Dependencies<a class="headerlink" href="#test-dependencies" title="Link to this heading">¶</a></h3>
<p>During a Bioconda package build, the stringent <code class="docutils literal notranslate"><span class="pre">mulled-build</span></code> test
runs the tests listed in the <code class="docutils literal notranslate"><span class="pre">test</span></code> section of the recipe’s <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code>
file in an environment which does not have the dependencies listed in the
<code class="docutils literal notranslate"><span class="pre">test:</span> <span class="pre">requires:</span></code> section. Because of this, all tests in that section
must only rely on runtime dependencies or the build will fail.</p>
<p>If you do want to run tests that rely on dependencies listed in <code class="docutils literal notranslate"><span class="pre">test:</span>
<span class="pre">requires:</span></code>, those tests should not be written directly in the <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code>
file but instead put in a <code class="docutils literal notranslate"><span class="pre">run_test.sh</span></code> file in the same directory as the
<code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code> file. This file is picked up and run by the standard conda build
test that occurs before the <code class="docutils literal notranslate"><span class="pre">mulled-build</span></code>, but is not passed on to and run
by the <code class="docutils literal notranslate"><span class="pre">mulled-build</span></code>.</p>
</section>
<section id="patching">
<span id="id5"></span><h3>Patching<a class="headerlink" href="#patching" title="Link to this heading">¶</a></h3>
<p>Some recipes require small patches to get the tests to pass, for example,
fixing hard-coded shebang lines (as described at
<a class="reference internal" href="troubleshooting.html#perl-or-python-not-found"><span class="std std-ref">/usr/bin/perl or /usr/bin/python not found</span></a>). Other patches are more extensive. When
patching a recipe, please first make an effort to fix the issue upstream and
document that effort in your pull request by either linking to the relevant
upstream PR or indicating that you have contacted the author. The goal is not
to block merging your PR until upstream is fixed, but rather to make sure
upstream authors know there’s an issue that other users (including non-bioconda
users) might be having. Ideally, upstream would fix the issue quickly and the
PR could be modified, but it’s fine to merge with the patches and if/when
upstream fixes, a separate bioconda PR could be opened that pulls in those
upstream changes.</p>
</section>
</section>
<section id="python">
<h2>Python<a class="headerlink" href="#python" title="Link to this heading">¶</a></h2>
<p>If a Python package is available on PyPI, use <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">pypi</span>
<span class="pre">&lt;packagename&gt;</span></code> , or the <a class="reference external" href="https://github.com/conda-incubator/grayskull">more recent alternative</a> <code class="docutils literal notranslate"><span class="pre">grayskull</span> <span class="pre">pypi</span> <span class="pre">&lt;packagename&gt;</span></code>
to create a recipe, then remove the <code class="docutils literal notranslate"><span class="pre">bld.bat</span></code> and any extra comments
in <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code> and <code class="docutils literal notranslate"><span class="pre">build.sh</span></code>. The test that is automatically
added is probably sufficient. The exception is when the package also installs
a command-line tool, in which case that should be tested as well.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">pypi</span></code> and <code class="docutils literal notranslate"><span class="pre">grayskull</span> <span class="pre">pypi</span></code> only work with packages that have a source
distribution produced with <code class="docutils literal notranslate"><span class="pre">python</span> <span class="pre">setup.py</span> <span class="pre">sdist</span></code>. Packages on PyPI
which only have a wheel will not work.
Packages containing only “built source” distributions produced with
<code class="docutils literal notranslate"><span class="pre">python</span> <span class="pre">setup.py</span> <span class="pre">bdist</span></code> on UNIX will likewise not work.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Make sure you have a conda-build version 3.x when running
<code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">pypi</span> <span class="pre">&lt;packagename&gt;</span></code>. If you are still using conda-build
2.x, either update your conda-build package, or follow the migration
guidelines in <span class="xref std std-ref">cb3-main</span>.</p>
</div>
<ul class="simple">
<li><p>typical <code class="docutils literal notranslate"><span class="pre">import</span></code> check: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/pysam">pysam</a></p></li>
<li><p>import and command-line tests: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/chanjo">chanjo</a></p></li>
</ul>
<p>By default, Python recipes (those that have <code class="docutils literal notranslate"><span class="pre">python</span></code> listed as a dependency)
must be successfully built and tested on all supported Python versions in order to
pass. However, many Python packages are not fully compatible across all Python
versions. Use the <code class="xref any docutils literal notranslate"><span class="pre">preprocessing</span> <span class="pre">selectors</span></code>
in the meta.yaml file along with the <code class="docutils literal notranslate"><span class="pre">build/skip</span></code> entry to indicate that
a recipe should be skipped.</p>
<p>For example, a recipe that only runs on Python 2.7 should include the
following:</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">host</span><span class="p">:</span>
<span class="w">  </span><span class="p p-Indicator">-</span><span class="w">   </span><span class="l l-Scalar l-Scalar-Plain">python &lt;3</span>
</pre></div>
</div>
<p>Or a package that only runs on Python 3.6 and 3.7:</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">host</span><span class="p">:</span>
<span class="w">  </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">python &gt;=3.6</span>
</pre></div>
</div>
<p>Alternatively, for straightforward compatibility fixes you can apply a <code class="xref any docutils literal notranslate"><span class="pre">patch</span>
<span class="pre">in</span> <span class="pre">the</span> <span class="pre">meta.yaml</span></code>.</p>
</section>
<section id="r-cran">
<span id="id6"></span><h2>R (CRAN)<a class="headerlink" href="#r-cran" title="Link to this heading">¶</a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Most R packages on CRAN should be submitted at Conda-Forge.
Specifically, if the CRAN package has a Bioconductor package dependency, it belongs in
Bioconda. If the CRAN package does not have a Bioconductor package dependency, it
belongs in Conda-Forge.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Using the <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">cran</span></code> method results in a recipe intended to be
built for Windows as well, with lines like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="o">%</span> <span class="nb">set</span> <span class="n">posix</span> <span class="o">=</span> <span class="s1">&#39;m2-&#39;</span> <span class="k">if</span> <span class="n">win</span> <span class="k">else</span> <span class="s1">&#39;&#39;</span> <span class="o">%</span><span class="p">}</span>
<span class="p">{</span><span class="o">%</span> <span class="nb">set</span> <span class="n">native</span> <span class="o">=</span> <span class="s1">&#39;m2w64-&#39;</span> <span class="k">if</span> <span class="n">win</span> <span class="k">else</span> <span class="s1">&#39;&#39;</span> <span class="o">%</span><span class="p">}</span>
</pre></div>
</div>
<p>and</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="w">   </span><span class="nt">test</span><span class="p">:</span>
<span class="w">     </span><span class="nt">commands</span><span class="p">:</span>
<span class="w">       </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">$R -e &quot;library(&#39;RNeXML&#39;)&quot;</span><span class="w">  </span><span class="c1"># [not win]</span>
<span class="w">       </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="s">&quot;\&quot;%R%\&quot;</span><span class="nv"> </span><span class="s">-e</span><span class="nv"> </span><span class="s">\&quot;library(&#39;RNeXML&#39;)\&quot;&quot;</span><span class="w">  </span><span class="c1"># [win]</span>
<span class="l l-Scalar l-Scalar-Plain">and</span>

<span class="l l-Scalar l-Scalar-Plain">.. code-block:</span><span class="p p-Indicator">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">yaml</span>

<span class="w">    </span><span class="l l-Scalar l-Scalar-Plain">requirements</span><span class="p p-Indicator">:</span>
<span class="w">      </span><span class="nt">build</span><span class="p">:</span>
<span class="w">        </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="p p-Indicator">{{</span><span class="w"> </span><span class="nv">posix</span><span class="w"> </span><span class="p p-Indicator">}}</span><span class="l l-Scalar l-Scalar-Plain">zip</span><span class="w">               </span><span class="c1"># [win]</span>
</pre></div>
</div>
<p>The bioconda channel does not build for Windows. To keep recipes
streamlined, please remove the “set posix” and “set native” lines described
above, remove the whole <code class="docutils literal notranslate"><span class="pre">build</span></code> entry under <code class="docutils literal notranslate"><span class="pre">requirements</span></code>, and convert
the <code class="docutils literal notranslate"><span class="pre">test:commands:</span></code> block to only:</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">test</span><span class="p">:</span>
<span class="w">  </span><span class="nt">commands</span><span class="p">:</span>
<span class="w">    </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">$R -e &quot;library(&#39;RNeXML&#39;)&quot;</span>
</pre></div>
</div>
</div>
<p>Use <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">cran</span> <span class="pre">&lt;packagename&gt;</span></code> where <code class="docutils literal notranslate"><span class="pre">packagename</span></code> is a
package available on CRAN and is <em>case-sensitive</em>. Either run that command
in the <code class="docutils literal notranslate"><span class="pre">recipes</span></code> dir or move the recipe it creates to <code class="docutils literal notranslate"><span class="pre">recipes</span></code>. The
recipe name will have an <code class="docutils literal notranslate"><span class="pre">r-</span></code> prefix and will be converted to
lowercase. Typically can be used without modification, though
dependencies may also need recipes. For further details on skeleton entries, you
can also refer to the <a class="reference external" href="https://github.com/conda/conda-build/blob/master/conda_build/skeletons/cran.py">cran skeleton template</a>.</p>
<p>Please remove any unnecessary comments and delete the <code class="docutils literal notranslate"><span class="pre">bld.bat</span></code> file which is
used only on Windows.</p>
<p>If the recipe was created using <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">cran</span></code> or the
<code class="docutils literal notranslate"><span class="pre">scripts/bioconductor_skeleton.py</span></code> script, the default test is
probably sufficient. Otherwise see the examples below to see how tests are
performed for R packages.</p>
<ul class="simple">
<li><p>recipe for R package not on CRAN, also with patch: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/r-spp">spp</a></p></li>
</ul>
</section>
<section id="r-bioconductor">
<h2>R (Bioconductor)<a class="headerlink" href="#r-bioconductor" title="Link to this heading">¶</a></h2>
<p>Use the <code class="docutils literal notranslate"><span class="pre">bioconda-utils</span> <span class="pre">bioconductor-skeleton</span></code> tool to build a Bioconductor
skeleton. After using the <span class="xref std std-ref">bootstrap</span> method to set up a testing
environment and activating that environment (which will ensure the correct
versions of bioconda-utils and conda-build), from the top level of the
<code class="docutils literal notranslate"><span class="pre">bioconda-recipes</span></code> repository run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bioconda</span><span class="o">-</span><span class="n">utils</span> <span class="n">bioconductor</span><span class="o">-</span><span class="n">skeleton</span> <span class="n">recipes</span> <span class="n">config</span><span class="o">.</span><span class="n">yml</span> <span class="n">DESeq2</span>
</pre></div>
</div>
<p>Note that the provided package name is a case-sensitive package available on
Bioconductor. The output recipe name will have a <code class="docutils literal notranslate"><span class="pre">bioconductor-</span></code> prefix and
will be converted to lowercase.  Data packages will be detected automatically,
and a post-link script (see <a class="reference external" href="https://github.com/bioconda/bioconda-utils/pull/169">https://github.com/bioconda/bioconda-utils/pull/169</a>
for details). Typically the resulting recipe can be used without modification,
though dependencies may also need recipes. Recipes for dependencies with an
<code class="docutils literal notranslate"><span class="pre">r-</span></code> prefix should be created at Conda-Forge unless the CRAN package has a
Bioconductor dependency; see <a class="reference internal" href="#r-cran"><span class="std std-ref">R (CRAN)</span></a> above.</p>
<ul class="simple">
<li><p>typical bioconductor recipe: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/bioconductor-limma">bioconductor-limma/meta.yaml</a></p></li>
</ul>
</section>
<section id="r-other-sources">
<h2>R (other sources)<a class="headerlink" href="#r-other-sources" title="Link to this heading">¶</a></h2>
<p>If a package is only provided in a public repository (e.g. at github or
bitbucket) or via some other website, first check with the authors of the
package, if they are planning to publish it on CRAN or Bioconductor. This is
always preferable, as it will ensure quality control and permanent availability
at a stable URL, and can warrant waiting for such a publication. If this is not
planned, you should check if a tagged version is available in a public repo (see
<a class="reference internal" href="#stable-url"><span class="std std-ref">infos on stable URLs</span></a> above) or if the authors are willing
to generate one. Only if none of this succeeds, the risk of the source
repository or website disappearing should be taken.</p>
<p>Once you have obtained a <a class="reference internal" href="#stable-url"><span class="std std-ref">stable URL</span></a> to the package, follow
the <a class="reference internal" href="#r-cran"><span class="std std-ref">guidelines for R packages on CRAN</span></a> above and adjust the URL
and checksum accordingly.</p>
</section>
<section id="java">
<h2>Java<a class="headerlink" href="#java" title="Link to this heading">¶</a></h2>
<p>Add a wrapper script if the software is typically called via <code class="docutils literal notranslate"><span class="pre">java</span> <span class="pre">-jar</span> <span class="pre">...</span></code>.
Sometimes the software already comes with one; for example, <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/fastqc">fastqc</a>
already had a wrapper script, but <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/peptide-shaker">peptide-shaker</a>
did not.</p>
<p>New recipes should use the <code class="docutils literal notranslate"><span class="pre">openjdk</span></code> package from conda-forge <a class="reference external" href="https://github.com/conda-forge/openjdk-feedstock">(recipe feedstock)</a>,
the java-jdk package from bioconda is deprecated.</p>
<p>JAR files should go in <code class="docutils literal notranslate"><span class="pre">$PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM</span></code>.
A wrapper script should be placed here as well, and symlinked to
<code class="docutils literal notranslate"><span class="pre">$PREFIX/bin</span></code>.</p>
<ul class="simple">
<li><p>Example with added wrapper script: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/peptide-shaker">peptide-shaker</a></p></li>
<li><p>Example with patch to fix memory: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/fastqc">fastqc</a></p></li>
</ul>
</section>
<section id="perl">
<h2>Perl<a class="headerlink" href="#perl" title="Link to this heading">¶</a></h2>
<p>Use <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">cpan</span> <span class="pre">&lt;packagename&gt;</span></code> to build a recipe for Perl and
place the recipe in the <code class="docutils literal notranslate"><span class="pre">recipes</span></code> dir. The recipe will have the
<code class="docutils literal notranslate"><span class="pre">perl-</span></code> prefix.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Historically, before conda-forge hosted Perl packages, many
non-bioinformatics-related Perl packages were hosted on Bioconda. These are
slowly being migrated to conda-forge.</p>
</div>
<p>An example of such a package is
<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/perl-module-build">perl-module-build</a>.</p>
<p>Alternatively, you can additionally ensure the build requirements for
the recipe include <code class="docutils literal notranslate"><span class="pre">perl-app-cpanminus</span></code>, and then the <code class="docutils literal notranslate"><span class="pre">build.sh</span></code>
script can be simplified. An example of this simplification is
<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/perl-time-hires">perl-time-hires</a>.</p>
<p>If the recipe was created with <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">cpan</span></code>, the tests are
likely sufficient. Otherwise, test the import of modules (see the
<code class="docutils literal notranslate"><span class="pre">imports</span></code> section of the <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code> files in above examples).</p>
<p>Additionally, if the recipe was created with <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">skeleton</span> <span class="pre">cpan</span></code>, several modifications
are necessary to satisfy bioconda policies:</p>
<ul class="simple">
<li><p>remove the <code class="docutils literal notranslate"><span class="pre">bld.bat</span></code> script</p></li>
<li><p>remove the <code class="docutils literal notranslate"><span class="pre">source/fn</span></code> entry in <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code></p></li>
<li><p>the <code class="docutils literal notranslate"><span class="pre">requirements/build</span></code> keyword in <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code> should be changed to
<code class="docutils literal notranslate"><span class="pre">requirements/host</span></code></p></li>
</ul>
</section>
<section id="c-c">
<h2>C/C++<a class="headerlink" href="#c-c" title="Link to this heading">¶</a></h2>
<p>Build tools (e.g., <code class="docutils literal notranslate"><span class="pre">autoconf</span></code>) and compilers (e.g., <code class="docutils literal notranslate"><span class="pre">gcc</span></code>) should be
specified in the build requirements. Compilers are handled via a special macro.
E.g., <code class="docutils literal notranslate"><span class="pre">{{</span> <span class="pre">compiler('c')}}</span></code> ensures that the correct version of <code class="docutils literal notranslate"><span class="pre">gcc</span></code> is used.
For the C++ variant <code class="docutils literal notranslate"><span class="pre">g++</span></code>, you need to use <code class="docutils literal notranslate"><span class="pre">{{</span> <span class="pre">compiler('cxx')</span> <span class="pre">}}</span></code>.
These rules apply for both Linux and macOS.</p>
<p>Conda distinguishes between dependencies needed for building (the <code class="docutils literal notranslate"><span class="pre">build</span></code> section),
and dependencies needed during build time (the <code class="docutils literal notranslate"><span class="pre">host</span></code> section).
For example, the following</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">requirements</span><span class="p">:</span>
<span class="w">  </span><span class="nt">build</span><span class="p">:</span>
<span class="w">    </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="p p-Indicator">{{</span><span class="w"> </span><span class="nv">compiler(&#39;c&#39;)</span><span class="w"> </span><span class="p p-Indicator">}}</span>
<span class="w">  </span><span class="nt">host</span><span class="p">:</span>
<span class="w">    </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">zlib</span>
<span class="w">  </span><span class="nt">run</span><span class="p">:</span>
<span class="w">    </span><span class="p p-Indicator">-</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">zlib</span>
</pre></div>
</div>
<p>specifies that a recipe needs the C compiler to build, and zlib present during
building and running.</p>
<p>For two examples see:</p>
<ul class="simple">
<li><p>example requiring <code class="docutils literal notranslate"><span class="pre">autoconf</span></code>: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/srprism">srprism</a></p></li>
<li><p>simple example: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/samtools">samtools</a></p></li>
</ul>
<p>If the package uses <code class="docutils literal notranslate"><span class="pre">zlib</span></code>, then please see the <a class="reference internal" href="troubleshooting.html#zlib"><span class="std std-ref">troubleshooting section on zlib</span></a>.</p>
<p>If your package links dynamically against a particular library, it is
often necessary to pin the version against which it was compiled, in
order to avoid ABI incompatibilities. Instead of hardcoding a particular
version in the recipe, we rely on conda doing this automatically.
We use globally defined configurations, namely <a class="reference external" href="https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml">this for dependencies from conda-forge</a>
and <a class="reference external" href="https://github.com/bioconda/bioconda-utils/blob/master/bioconda_utils/bioconda_utils-conda_build_config.yaml">this for dependencies in bioconda</a>.
If you need to pin another library, please notify &#64;bioconda/core, and we will extend these lists.</p>
<p>It’s not uncommon to have difficulty compiling package into a portable
conda package. Since there is no single solution, here are some examples
of how bioconda contributors have solved compiling issues to give you
some ideas on what to try:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/ococo">ococo</a>
edits the source in <code class="docutils literal notranslate"><span class="pre">build.sh</span></code> to accommodate the C++ compiler on OSX</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/muscle">muscle</a>
patches the makefile on OSX so it doesn’t use static libs</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/metavelvet">metavelvet</a>,
<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/eautils">eautils</a>,
<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/preseq">preseq</a>
have several patches to their makefiles to fix <code class="docutils literal notranslate"><span class="pre">LIBS</span></code> and <code class="docutils literal notranslate"><span class="pre">INCLUDES</span></code>,
<code class="docutils literal notranslate"><span class="pre">INCLUDEARGS</span></code>, and <code class="docutils literal notranslate"><span class="pre">CFLAGS</span></code></p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/mapsplice">mapsplice</a>
includes an older version of samtools; the included samtools’ makefile is
patched to work in conda envs.</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/mosaik">mosaik</a>
has platform-specific patches – one removes <code class="docutils literal notranslate"><span class="pre">-static</span></code> on linux, and the
other sets <code class="docutils literal notranslate"><span class="pre">BLD_PLATFORM</span></code> correctly on OSX</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/mothur">mothur</a>
and <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/soapdenovo">soapdenovo</a>
have many fixes to makefiles</p></li>
</ul>
</section>
<section id="haskell">
<h2>Haskell<a class="headerlink" href="#haskell" title="Link to this heading">¶</a></h2>
<p>Bioconda has a small number of haskell tools. Most often they are built with
<code class="docutils literal notranslate"><span class="pre">stack</span></code> (which is available on <a class="reference external" href="https://github.com/conda-forge/stack-feedstock">conda-forge</a>). <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/blob/master/recipes/ngless/build.sh">NGLess</a>
provides an example of how to call <code class="docutils literal notranslate"><span class="pre">stack</span></code>. Here are a few notes:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">LD_LIBRARY_PATH</span></code>/<code class="docutils literal notranslate"><span class="pre">LIBRARY_PATH</span></code> are set to include both
<code class="docutils literal notranslate"><span class="pre">${PREFIX}/lib</span></code> and the system paths (otherwise, <code class="docutils literal notranslate"><span class="pre">stack</span> <span class="pre">setup</span></code> will
fail).</p></li>
<li><p>Create a directory (called <code class="docutils literal notranslate"><span class="pre">fake-home</span></code> in this example) and set it as
<code class="docutils literal notranslate"><span class="pre">$HOME</span></code>, further setting <code class="docutils literal notranslate"><span class="pre">$STACK_ROOT</span></code> to use a subdirectory of this
<code class="docutils literal notranslate"><span class="pre">$HOME</span></code>.</p></li>
</ul>
<p>Mac OS X support is generally missing (any help is appreciated, see <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/issues/6607">#6607</a>).</p>
</section>
<section id="general-command-line-tools">
<h2>General command-line tools<a class="headerlink" href="#general-command-line-tools" title="Link to this heading">¶</a></h2>
<p>If a command-line tool is installed, it should be tested. If it has a
shebang line, it should be patched to use <code class="docutils literal notranslate"><span class="pre">/usr/bin/env</span></code> for more
general use. An example of this is <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/fastq-screen">fastq-screen</a>.</p>
<p>For command-line tools, running the program with no arguments, checking
the programs version (e.g. with <code class="docutils literal notranslate"><span class="pre">-v</span></code>) or checking the command-line
help is sufficient if doing so returns an exit code 0. Often the output
is piped to <code class="docutils literal notranslate"><span class="pre">/dev/null</span></code> to avoid output during recipe builds.</p>
<p>Examples:</p>
<ul class="simple">
<li><p>exit code 0: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/bedtools">bedtools</a></p></li>
<li><p>exit code 255 in a separate script: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/ucsc-bedgraphtobigwig">ucsc-bedgraphtobigwig</a></p></li>
<li><p>confirm expected text in stderr: <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/weblogo">weblogo</a></p></li>
</ul>
<p>If a package depends on Python and has a custom build string, then
<code class="docutils literal notranslate"><span class="pre">py{{CONDA_PY}}</span></code> must be contained in that build string. Otherwise Python
will be automatically pinned to one minor version, resulting in dependency
conflicts with other packages. See <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/mapsplice">mapsplice</a>
for an example of this.</p>
</section>
<section id="metapackages">
<h2>Metapackages<a class="headerlink" href="#metapackages" title="Link to this heading">¶</a></h2>
<p><a class="reference external" href="https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/packages.html#metapackages">Metapackages</a> tie
together other packages. All they do is define dependencies. For example, the
<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/hubward-all">hubward-all</a>
metapackage specifies the various other conda packages needed to get full
<code class="docutils literal notranslate"><span class="pre">hubward</span></code> installation running just by installing one package. Other
metapackages might tie together conda packages with a theme. For example, all
UCSC utilities related to bigBed files, or a set of packages useful for variant
calling.</p>
<p>For packages that are not anchored to a particular package (as in the last
example above), we recommended <a class="reference external" href="http://semver.org/">semantic versioning</a>
starting at 1.0.0 for metapackages.</p>
</section>
<section id="other-examples-of-interest">
<h2>Other examples of interest<a class="headerlink" href="#other-examples-of-interest" title="Link to this heading">¶</a></h2>
<p>Packaging is hard. Here are some examples, in no particular order, of how
contributors have solved various problems:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/graphviz">graphviz</a>
has an OS-specific option to <code class="docutils literal notranslate"><span class="pre">configure</span></code></p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/crossmap">crossmap</a>
removes libs that are shipped with the source distribution</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/hisat2">hisat2</a>
runs <code class="docutils literal notranslate"><span class="pre">2to3</span></code> to make it Python 3 compatible, and copies over individual
scripts to the bin dir</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/krona">krona</a>
has a <code class="docutils literal notranslate"><span class="pre">post-link.sh</span></code> script that gets called after installation to alert
the user a manual step is required</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/htslib">htslib</a>
has a small test script that creates example data and runs multiple programs
on it</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/spectacle">spectacle</a>
runs <code class="docutils literal notranslate"><span class="pre">2to3</span></code> to make the wrapper script Python 3 compatible, patches the
wrapper script to have a shebang line, deletes example data to avoid taking
up space in the bioconda channel, and includes a script for downloading the
example data separately.</p></li>
<li><p><a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/gatk">gatk</a> is
a package for licensed software that cannot be redistributed. The package
installs a placeholder script (in this case doubling as the <code class="docutils literal notranslate"><span class="pre">jar</span></code> <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/blob/master/GUIDELINES.md#java">wrapper</a>)
to alert the user if the program is not installed, along with a separate
script (<code class="docutils literal notranslate"><span class="pre">gatk-register</span></code>) to copy in a user-supplied archive/binary to the
conda environment</p></li>
</ul>
</section>
<section id="name-collisions-identical-names">
<h2>Name collisions (identical names)<a class="headerlink" href="#name-collisions-identical-names" title="Link to this heading">¶</a></h2>
<p>In rare cases, a new recipe may have an identical name as an existing conda, conda-forge, bioconda, or Python package. This sort of naming collision should be avoided. For example, the <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/weblogo">weblogo</a>
recipe is for the standard command-line tool. There is also a Python package
called <code class="docutils literal notranslate"><span class="pre">weblogo</span></code> <a class="reference external" href="https://pypi.python.org/pypi/weblogo">on PyPI</a>. In this case, to reduce ambiguity
we prefixed the Python package with <code class="docutils literal notranslate"><span class="pre">python-</span></code> (see <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/python-weblogo">python-weblogo</a>).</p>
<p>If in doubt about how to handle a naming collision, please submit an
issue, or ask <code class="docutils literal notranslate"><span class="pre">&#64;bioconda/core</span></code> in a PR.</p>
</section>
<section id="tests">
<span id="id12"></span><h2>Tests<a class="headerlink" href="#tests" title="Link to this heading">¶</a></h2>
<p>An adequate test must be included in the recipe. An “adequate” test
depends on the recipe, but must be able to detect a successful
installation. While many packages may ship their own test suite (unit
tests or otherwise), including these in the recipe is not recommended
since it may timeout the build system on CircleCI. We especially want to avoid
including any kind of test data in the repository.</p>
<p>Note that a test must return an exit code of 0. The test can be in the <code class="docutils literal notranslate"><span class="pre">test</span></code>
field of <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code>, or can be a separate script (see the <a class="reference external" href="https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html?highlight=test-section#test-section">relevant conda
docs</a> for
testing).</p>
<p>It is recommended to pipe unneeded stdout/stderr to /dev/null to avoid
cluttering the output in the CircleCI build environment.</p>
</section>
<section id="link-and-unlink-scripts-pre-and-post-install-hooks">
<h2>Link and unlink scripts (pre- and post- install hooks)<a class="headerlink" href="#link-and-unlink-scripts-pre-and-post-install-hooks" title="Link to this heading">¶</a></h2>
<p>It is possible to include <a class="reference external" href="https://docs.conda.io/projects/conda-build/en/latest/resources/link-scripts.html#adding-pre-link-post-link-and-pre-unlink-scripts">scripts</a> that are
executed before or after installing a package, or before uninstalling
a package. These scripts can be helpful for alerting the user that manual
actions are required after adding or removing a package. For example,
a <code class="docutils literal notranslate"><span class="pre">post-link.sh</span></code> script may be used to alert the user that he or she will
need to create a database or modify a settings file. Any package that requires
a manual preparatory step before it can be used should consider alerting the
user via an <code class="docutils literal notranslate"><span class="pre">echo</span></code> statement in a <code class="docutils literal notranslate"><span class="pre">post-link.sh</span></code> script. These scripts may
be added at the same level as <code class="docutils literal notranslate"><span class="pre">meta.yaml</span></code> and <code class="docutils literal notranslate"><span class="pre">build.sh</span></code>:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">pre-link.sh</span></code> is executed <em>prior</em> to linking (installation). An error
causes conda to stop.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">post-link.sh</span></code> is executed <em>after</em> linking (installation). When the
post-link step fails, no package metadata is written, and the package is not
considered installed.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">pre-unlink.sh</span></code> is executed <em>prior</em> to unlinking (uninstallation). Errors
are ignored. Used for cleanup.</p></li>
</ul>
<p>These scripts have access to the following environment variables:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">$PREFIX</span></code> The install prefix</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">$PKG_NAME</span></code> The name of the package</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">$PKG_VERSION</span></code> The version of the package</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">$PKG_BUILDNUM</span></code> The build number of the package</p></li>
</ul>
</section>
<section id="versions">
<h2>Versions<a class="headerlink" href="#versions" title="Link to this heading">¶</a></h2>
<p>In general, recipes can be updated in-place. The older package[s] will continue
to be hosted and available on anaconda.org while the recipe will reflect just
the most recent package.</p>
<p>However, if an older version of a packages is required but has not yet had
a package built, create a subdirectory of the recipe named after the old
version and put the recipe there. Examples of this can be found in <a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/bowtie2">bowtie2</a>,
<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/bx-python">bx-python</a>,
and others.</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<p class="logo">
  <a href="../index.html">
    <img class="logo" src="../_static/logo/bioconda_monochrome_small.png" alt="Logo" />
    
  </a>
</p>








<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../faqs.html">FAQs</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="index.html">Contributing to Bioconda</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="setup.html">Initial Setup</a></li>
<li class="toctree-l2"><a class="reference internal" href="workflow.html">Contribution Workflow</a></li>
<li class="toctree-l2"><a class="reference internal" href="building-locally.html">Testing Recipes Locally</a></li>
<li class="toctree-l2"><a class="reference internal" href="troubleshooting.html">Troubleshooting</a></li>
<li class="toctree-l2"><a class="reference internal" href="build-system.html">Build system</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Guidelines for <code class="docutils literal notranslate"><span class="pre">bioconda</span></code> recipes</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#bioconda-recipe-checklist">bioconda recipe checklist</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#stable-urls">Stable urls</a></li>
<li class="toctree-l4"><a class="reference internal" href="#hashes">Hashes</a></li>
<li class="toctree-l4"><a class="reference internal" href="#build-numbers">Build numbers</a></li>
<li class="toctree-l4"><a class="reference internal" href="#bat-files"><code class="docutils literal notranslate"><span class="pre">.bat</span></code> files</a></li>
<li class="toctree-l4"><a class="reference internal" href="#comments-in-recipes">Comments in recipes</a></li>
<li class="toctree-l4"><a class="reference internal" href="#filesystem-hierarchy-standard">Filesystem Hierarchy Standard</a></li>
<li class="toctree-l4"><a class="reference internal" href="#existing-package-exceptions">Existing package exceptions</a></li>
<li class="toctree-l4"><a class="reference internal" href="#packages-appropriate-for-bioconda">Packages appropriate for bioconda</a></li>
<li class="toctree-l4"><a class="reference internal" href="#noarch-packages">“Noarch” packages</a></li>
<li class="toctree-l4"><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li class="toctree-l4"><a class="reference internal" href="#test-dependencies">Test Dependencies</a></li>
<li class="toctree-l4"><a class="reference internal" href="#patching">Patching</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#python">Python</a></li>
<li class="toctree-l3"><a class="reference internal" href="#r-cran">R (CRAN)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#r-bioconductor">R (Bioconductor)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#r-other-sources">R (other sources)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#java">Java</a></li>
<li class="toctree-l3"><a class="reference internal" href="#perl">Perl</a></li>
<li class="toctree-l3"><a class="reference internal" href="#c-c">C/C++</a></li>
<li class="toctree-l3"><a class="reference internal" href="#haskell">Haskell</a></li>
<li class="toctree-l3"><a class="reference internal" href="#general-command-line-tools">General command-line tools</a></li>
<li class="toctree-l3"><a class="reference internal" href="#metapackages">Metapackages</a></li>
<li class="toctree-l3"><a class="reference internal" href="#other-examples-of-interest">Other examples of interest</a></li>
<li class="toctree-l3"><a class="reference internal" href="#name-collisions-identical-names">Name collisions (identical names)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#tests">Tests</a></li>
<li class="toctree-l3"><a class="reference internal" href="#link-and-unlink-scripts-pre-and-post-install-hooks">Link and unlink scripts (pre- and post- install hooks)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#versions">Versions</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="linting.html">Linting</a></li>
<li class="toctree-l2"><a class="reference internal" href="updating.html">Updating recipes</a></li>
<li class="toctree-l2"><a class="reference internal" href="updating.html#updating-recipes-for-a-pinning-change">Updating recipes for a pinning change</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../developer/index.html">Developer Docs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorials/index.html">Tutorials</a></li>
</ul>

<ul>
  
  <li class="toctree-l1"><a href="../conda-package_index.html">Browse packages</a></li>
  
  <li class="toctree-l1"><a href="https://github.com/bioconda/bioconda-recipes">Bioconda @ Github</a></li>
  
  <li class="toctree-l1"><a href="https://gitter.im/bioconda/Lobby"><img alt="Gitter" src="https://img.shields.io/gitter/room/bioconda/Lobby.svg"></a></li>
  
</ul>
<div id="searchbox" style="display: none" role="search">
    <h3 id="searchlabel">Search</h3>
    <p id="searchdesc">packages & docs</p>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div><script>
    let closest;
    // Get the last part of the URL after the # symbol - should match the div id to display
    let id = window.location.hash.substring(1);
    let elem = document.getElementById(id);
    if (elem != null) {
        // Get the closest "details" element and open it
        closest = elem.closest("details");
        if (closest) closest.open = true;
    }
</script>

    
    <div class="footer">
      &#169;2016-2024, The Bioconda Team.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.0.2</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="../_sources/contributor/guidelines.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    <div class="git-ribbon">
      <a href="https://github.com/bioconda/bioconda-docs/edit/main/source/contributor/guidelines.rst" rel="me">Edit me on GitHub</a>
    </div>
  </body>
</html>