
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>Package Recipe &#39;bioconductor-fraser&#39; &#8212; Bioconda  documentation</title>
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/alabaster.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/graphviz.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/style.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/font-awesome-4.7.0/css/font-awesome.min.css" />
    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/vega.min.js"></script>
    <script src="../../_static/vega-lite.min.js"></script>
    <script src="../../_static/vega-embed.min.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link href="https://fonts.googleapis.com/css?family=Lato|Raleway" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Raleway:wght@700&display=swap" rel="stylesheet">


    <link rel="apple-touch-icon" sizes="57x57" href="../../_static/apple-icon-57x57.png">
    <link rel="apple-touch-icon" sizes="60x60" href="../../_static/apple-icon-60x60.png">
    <link rel="apple-touch-icon" sizes="72x72" href="../../_static/apple-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="76x76" href="../../_static/apple-icon-76x76.png">
    <link rel="apple-touch-icon" sizes="114x114" href="../../_static/apple-icon-114x114.png">
    <link rel="apple-touch-icon" sizes="120x120" href="../../_static/apple-icon-120x120.png">
    <link rel="apple-touch-icon" sizes="144x144" href="../../_static/apple-icon-144x144.png">
    <link rel="apple-touch-icon" sizes="152x152" href="../../_static/apple-icon-152x152.png">
    <link rel="apple-touch-icon" sizes="180x180" href="../../_static/apple-icon-180x180.png">
    <link rel="icon" type="image/png" sizes="192x192"  href="../../_static/android-icon-192x192.png">
    <link rel="icon" type="image/png" sizes="32x32" href="../../_static/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="96x96" href="../../_static/favicon-96x96.png">
    <link rel="icon" type="image/png" sizes="16x16" href="../../_static/favicon-16x16.png">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="msapplication-TileImage" content="../../_static/ms-icon-144x144.png">

    <link rel="manifest" href="/manifest.json">
    <meta name="theme-color" content="#ffffff">

   
  <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
  

  
  


  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="bioconductor-fraser">
<dl class="conda recipe">
<dt class="sig sig-object conda">
<em class="property"><span class="pre">recipe</span> </em><span class="sig-name descname"><span class="pre">bioconductor-fraser</span></span></dt>
<dd><p>Find RAre Splicing Events in RNA-Seq Data</p>
<dl class="field-list simple">
<dt class="field-odd">Homepage<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference external" href="https://bioconductor.org/packages/3.18/bioc/html/FRASER.html">https://bioconductor.org/packages/3.18/bioc/html/FRASER.html</a></p>
</dd>
<dt class="field-even">License<span class="colon">:</span></dt>
<dd class="field-even"><p>MIT + file LICENSE</p>
</dd>
<dt class="field-odd">Recipe<span class="colon">:</span></dt>
<dd class="field-odd"><p>/<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/bioconductor-fraser">bioconductor-fraser</a>/<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/bioconductor-fraser/meta.yaml">meta.yaml</a></p>
</dd>
<dt class="field-even">Links<span class="colon">:</span></dt>
<dd class="field-even"><p>https: <a href="#id2"><span class="problematic" id="id3">:https:`//doi.org/10.1038/s41467-020-20573-7`</span></a></p>
</dd>
</dl>
<p>Detection of rare aberrant splicing events in transcriptome profiles. Read count ratio expectations are modeled by an autoencoder to control for confounding factors in the data. Given these expectations, the ratios are assumed to follow a beta-binomial distribution with a junction specific dispersion. Outlier events are then identified as read-count ratios that deviate significantly from this distribution. FRASER is able to detect alternative splicing, but also intron retention. The package aims to support diagnostics in the field of rare diseases where RNA-seq is performed to identify aberrant splicing defects.</p>
</dd></dl>

<dl class="conda package">
<dt class="sig sig-object conda" id="package-bioconductor-fraser">
<em class="property"><span class="pre">package</span> </em><span class="sig-name descname"><span class="pre">bioconductor-fraser</span></span><a class="headerlink" href="#package-bioconductor-fraser" title="Permalink to this definition">¶</a></dt>
<dd><blockquote>
<div><p><a class="reference external" href="https://anaconda.org/bioconda/bioconductor-fraser"><img alt="(downloads)" src="https://img.shields.io/conda/dn/bioconda/bioconductor-fraser.svg?style=flat" /></a> <a class="reference external" href="https://quay.io/repository/biocontainers/bioconductor-fraser"><img alt="docker_bioconductor-fraser" src="https://quay.io/repository/biocontainers/bioconductor-fraser/status" /></a></p>
<dl class="field-list">
<dt class="field-odd">versions<span class="colon">:</span></dt>
<dd class="field-odd"><details><summary><span class="truncated-version-list"><code>1.14.0-0</code>,  <code>1.12.1-0</code>,  <code>1.10.0-1</code>,  <code>1.10.0-0</code>,  <code>1.6.1-2</code>,  <code>1.6.1-1</code>,  <code>1.6.1-0</code>,  <code>1.6.0-1</code>,  <code>1.6.0-0</code>,  </span></summary><p><code class="docutils literal notranslate"><span class="pre">1.14.0-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.12.1-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.10.0-1</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.10.0-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.6.1-2</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.6.1-1</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.6.1-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.6.0-1</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.6.0-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.4.0-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.2.1-1</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.2.1-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.2.0-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.0.1-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.0.0-0</span></code></p>
</details></dd>
<dt class="field-even">depends bioconductor-annotationdbi<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.64.0,&lt;1.65.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-annotationdbi<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.64.1,&lt;1.65.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-biobase<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.62.0,&lt;2.63.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-biobase<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.62.0,&lt;2.63.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-biocgenerics<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=0.48.0,&lt;0.49.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-biocgenerics<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=0.48.1,&lt;0.49.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-biocparallel<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.36.0,&lt;1.37.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-biocparallel<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.36.0,&lt;1.37.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-biomart<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.58.0,&lt;2.59.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-biomart<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.58.0,&lt;2.59.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-bsgenome<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.70.0,&lt;1.71.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-bsgenome<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.70.1,&lt;1.71.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-delayedarray<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=0.28.0,&lt;0.29.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-delayedarray<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=0.28.0,&lt;0.29.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-delayedmatrixstats<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.24.0,&lt;1.25.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-delayedmatrixstats<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.24.0,&lt;1.25.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-genomeinfodb<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.38.0,&lt;1.39.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-genomeinfodb<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.38.1,&lt;1.39.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-genomicalignments<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.38.0,&lt;1.39.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-genomicalignments<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.38.0,&lt;1.39.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-genomicfeatures<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.54.0,&lt;1.55.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-genomicfeatures<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.54.1,&lt;1.55.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-genomicranges<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.54.0,&lt;1.55.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-genomicranges<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.54.1,&lt;1.55.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-hdf5array<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.30.0,&lt;1.31.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-hdf5array<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.30.0,&lt;1.31.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-iranges<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.36.0,&lt;2.37.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-iranges<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.36.0,&lt;2.37.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-outrider<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.20.0,&lt;1.21.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-outrider<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.20.0,&lt;1.21.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-pcamethods<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.94.0,&lt;1.95.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-pcamethods<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.94.0,&lt;1.95.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-rhdf5<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.46.0,&lt;2.47.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-rhdf5<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.46.1,&lt;2.47.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-rsamtools<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.18.0,&lt;2.19.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-rsamtools<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.18.0,&lt;2.19.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-rsubread<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.16.0,&lt;2.17.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-rsubread<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=2.16.0,&lt;2.17.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-s4vectors<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=0.40.0,&lt;0.41.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-s4vectors<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=0.40.2,&lt;0.41.0a0</span></code></p>
</dd>
<dt class="field-even">depends bioconductor-summarizedexperiment<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.32.0,&lt;1.33.0</span></code></p>
</dd>
<dt class="field-odd">depends bioconductor-summarizedexperiment<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=1.32.0,&lt;1.33.0a0</span></code></p>
</dd>
<dt class="field-even">depends libblas<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=3.9.0,&lt;4.0a0</span></code></p>
</dd>
<dt class="field-odd">depends libgcc-ng<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=12</span></code></p>
</dd>
<dt class="field-even">depends liblapack<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=3.9.0,&lt;4.0a0</span></code></p>
</dd>
<dt class="field-odd">depends libstdcxx-ng<span class="colon">:</span></dt>
<dd class="field-odd"><p><code class="docutils literal notranslate"><span class="pre">&gt;=12</span></code></p>
</dd>
<dt class="field-even">depends r-base<span class="colon">:</span></dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">&gt;=4.3,&lt;4.4.0a0</span></code></p>
</dd>
<dt class="field-odd">depends r-bbmisc<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">depends r-cowplot<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
<dt class="field-odd">depends r-data.table<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">depends r-extradistr<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
<dt class="field-odd">depends r-generics<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">depends r-ggplot2<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
<dt class="field-odd">depends r-ggrepel<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">depends r-matrixstats<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
<dt class="field-odd">depends r-pheatmap<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">depends r-plotly<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
<dt class="field-odd">depends r-prroc<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">depends r-r.utils<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
<dt class="field-odd">depends r-rcolorbrewer<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">depends r-rcpp<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
<dt class="field-odd">depends r-rcpparmadillo<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">depends r-tibble<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
<dt class="field-odd">depends r-vgam<span class="colon">:</span></dt>
<dd class="field-odd"><p></p></dd>
<dt class="field-even">requirements<span class="colon">:</span></dt>
<dd class="field-even"><p></p></dd>
</dl>
<p class="rubric">Installation</p>
</div></blockquote>
<p>You need a conda-compatible package manager
(currently either <a class="reference external" href="https://mamba.readthedocs.io">micromamba</a>, <a class="reference external" href="https://mamba.readthedocs.io">mamba</a>, or <a class="reference external" href="https://docs.conda.io/projects/conda">conda</a>)
and the Bioconda channel already activated (see <span class="xref std std-ref">set-up-channels</span>).</p>
<p>While any of above package managers is fine, it is currently recommended to use either
micromamba or mamba (see <a class="reference external" href="https://mamba.readthedocs.io">here</a> for installation instructions).
We will show all commands using mamba below, but the arguments are the same for the two
others.</p>
<p>Given that you already have a conda environment in which you want to have this package, install with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>   <span class="n">mamba</span> <span class="n">install</span> <span class="n">bioconductor</span><span class="o">-</span><span class="n">fraser</span>

<span class="ow">and</span> <span class="n">update</span> <span class="k">with</span><span class="p">::</span>

   <span class="n">mamba</span> <span class="n">update</span> <span class="n">bioconductor</span><span class="o">-</span><span class="n">fraser</span>
</pre></div>
</div>
<p>To create a new environment, run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mamba</span> <span class="n">create</span> <span class="o">--</span><span class="n">name</span> <span class="n">myenvname</span> <span class="n">bioconductor</span><span class="o">-</span><span class="n">fraser</span>
</pre></div>
</div>
<p>with <code class="docutils literal notranslate"><span class="pre">myenvname</span></code> being a reasonable name for the environment
(see e.g. the <a class="reference external" href="https://mamba.readthedocs.io">mamba docs</a> for details and further options).</p>
<p>Alternatively, use the docker container:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>   docker pull quay.io/biocontainers/bioconductor-fraser:&lt;tag&gt;

(see `bioconductor-fraser/tags`_ for valid values for ``&lt;tag&gt;``)
</pre></div>
</div>
</dd></dl>

<script>
    var package = "bioconductor-fraser";
    var versions = ["1.14.0","1.12.1","1.10.0","1.10.0","1.6.1"];
</script><section id="download-stats">
<h2>Download stats<a class="headerlink" href="#download-stats" title="Permalink to this heading">¶</a></h2>
<div style="width: 100%" id="download_plot"></div>
<div style="width: 100%" id="cdf_plot"></div>

<script>
    window.onload = async function() {
        $.get(`https://raw.githubusercontent.com/bioconda/bioconda-plots/main/resources/cdf.vl.json`, function( data ) {
            let spec = JSON.parse(data);
            $.get(`https://raw.githubusercontent.com/bioconda/bioconda-plots/main/plots/cdf.json`, function( plot_data ) {
                $.get(`https://raw.githubusercontent.com/bioconda/bioconda-plots/main/plots/${package}/cdf.json`, function( single_point ) {
                    spec.data.values = JSON.parse(plot_data);
                    spec.data.values.push(JSON.parse(single_point).pop());
                    vegaEmbed('#cdf_plot', spec);
                });
            });
        });

        $.get(`https://raw.githubusercontent.com/bioconda/bioconda-plots/main/resources/versions.vl.json`, function( data ) {
            let spec = JSON.parse(data);
            $.get(`https://raw.githubusercontent.com/bioconda/bioconda-plots/main/plots/${package}/versions.json`, function( plot_data ) {
                spec.data.values = JSON.parse(plot_data);
                vegaEmbed('#download_plot', spec);
            });
        });
    }
</script>
</section>
<section id="link-to-this-page">
<h2>Link to this page<a class="headerlink" href="#link-to-this-page" title="Permalink to this heading">¶</a></h2>
<p>Render an <a class="reference external" href="http://bioconda.github.io/recipes/bioconductor-fraser/README.html"><img alt="install-with-bioconda" src="https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat" /></a> badge with the following MarkDown:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>[![install with bioconda](https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat)](http://bioconda.github.io/recipes/bioconductor-fraser/README.html)
</pre></div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<p class="logo">
  <a href="../../index.html">
    <img class="logo" src="../../_static/logo/bioconda_monochrome_small.png" alt="Logo" />
    
  </a>
</p>








<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../faqs.html">FAQs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contributor/index.html">Contributing to Bioconda</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../developer/index.html">Developer Docs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../tutorials/index.html">Tutorials</a></li>
</ul>

<ul>
  
  <li class="toctree-l1"><a href="../../conda-package_index.html">Browse packages</a></li>
  
  <li class="toctree-l1"><a href="https://github.com/bioconda/bioconda-recipes">Bioconda @ Github</a></li>
  
  <li class="toctree-l1"><a href="https://gitter.im/bioconda/Lobby"><img alt="Gitter" src="https://img.shields.io/gitter/room/bioconda/Lobby.svg"></a></li>
  
</ul>
<div id="searchbox" style="display: none" role="search">
    <h3 id="searchlabel">Search</h3>
    <p id="searchdesc">packages & docs</p>
    <div class="searchformwrapper">
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div><script>
    let closest;
    // Get the last part of the URL after the # symbol - should match the div id to display
    let id = window.location.hash.substring(1);
    let elem = document.getElementById(id);
    if (elem != null) {
        // Get the closest "details" element and open it
        closest = elem.closest("details");
        closest.open = true;
    }
</script>

    
    <div class="footer">
      &#169;2016-2024, The Bioconda Team.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 5.1.1</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="../../_sources/recipes/bioconductor-fraser/README.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    <div class="git-ribbon">
      <a href="https://github.com/bioconda/bioconda-recipes/edit/master/recipes/bioconductor-fraser/meta.yaml" rel="me">Edit me on GitHub</a>
    </div>
  </body>
</html>